### PR TITLE
Entitities and validation

### PR DESCRIPTION
**Pull request by anssi55**
This branch was about getting entities ready and validating the data going to database.

### CODE REVIEW

File: app/app.ts [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
+        this.makeConnection();
+    }
+
+}
+const app = new App();
+app.init();
+
+
+
+export default new App();
```
Comment (henrikra) [inline app/app.ts]: Hmm now there are two `App` instances 🤔

File: app/routes/index.ts [hunk 2]
```diff
@@ -25,19 +32,33 @@ export class Index {
     public notFound(req: Request, res: Response, next: NextFunction) {
         res.status(404).send({message:'Path not found'});
     }
+    public routePosts() {
+        const p = container.resolve<PostsRouter>('postsController');
+        this.router.route("/api/v1/posts")
```
Comment (henrikra) [inline app/routes/index.ts:37]: This is good!

File: app/routes/postsRouter.ts [hunk 3]
```diff
@@ -1,62 +1,85 @@
...
     }
+    //Delete a post from database
     public delete(req: Request, res: Response, next: NextFunction) {
         res.send("Nothing");
     }
+    
     init() {
-        this.router.get('/', this.getAll);
+        this.router.get('/api/v1/posts', this.getAll);
         this.router.get('/:id', this.getOne);
```
Comment (henrikra) [inline app/routes/postsRouter.ts]: Is all of this still needed?

File: package.json [hunk 4]
```diff
@@ -37,8 +41,11 @@
...
    },
    "devDependencies": {
-      "ts-node": "3.3.0",
       "@types/node": "^8.0.29",
-      "typescript": "3.3.3333"
+      "chai": "^4.2.0",
+      "mocha": "^6.1.4",
+      "ts-node": "^3.3.0",
+      "typescript": "3.3.3333",
+      "typings": "^2.1.1"
```
Comment (henrikra) [inline package.json]: What is this library used for?

File: package.json [hunk 5]
```diff
@@ -24,10 +24,14 @@
    },
    "homepage": "https://github.com/anssi55/Epic-bulletin-board-backend-project#readme",
    "dependencies": {
+      "@types/chai-http": "^4.2.0",
       "@types/express": "^4.17.0",
+      "@types/mocha": "^5.2.7",
```
Comment (henrikra) [inline package.json]: All types packages should be in devDependencies. When installing stuff to devDependencies you can do it like this `npm i -D @types/mocha`

File: test/posts.test.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+const expect = chai.expect;
+
+
+//Testing /posts-route
+
+describe('GET api/v1/posts', function()  {
+  it('responds with JSON array', async function()  {
+    return await chai.request().get('/posts')
+      .then(res => {
+        expect(res.status).to.eql(200)
```
Comment (henrikra) [inline test/posts.test.ts:16]: Good that you added some tests. Lets add more when Awilix is fully integrated 👍

File: tsconfig.json [hunk 7]
```diff
@@ -1,5 +1,7 @@
 {
    "compilerOptions": {
+      "resolveJsonModule": true,
```
Comment (henrikra) [inline tsconfig.json:3]: What part of the code needs this flag?

File: app/routes/index.ts [hunk 8]
```diff
@@ -25,19 +32,33 @@ export class Index {
...
+        .post(p.create)
+        this.router.route("/api/v1/posts/:id")
+        .get(p.getOne)
+        .put(p.update)
+        .delete(p.delete)
+    }
     
     //Routing all the addresses to right path
     init() {
+        container.resolve<PostsRouter>('postsController');
```
Comment (henrikra) [inline app/routes/index.ts]: Is this line needed?

File: app/routes/postsRouter.ts [hunk 9]
```diff
@@ -1,62 +1,85 @@
 'use strict';
-import {Router, Request, Response, NextFunction, response} from 'express';
-import {Posts} from "../orm/entities/Posts";
+import {Router, Request, Response, NextFunction} from 'express';
+import {Post} from "../orm/entities/Post";
 import {getRepository} from "typeorm";
+import {validate, IsInt, ValidationError} from "class-validator";
+import ErrorHandler from '../middleware/errorhandler';
+const awilix = require('awilix');
```
Comment (henrikra) [inline app/routes/postsRouter.ts]: Remove also all unused imports

**Review by henrikra — CHANGES_REQUESTED** [review]
Couple comments

File: tsconfig.json (hunk 7 above)
Comment (anssi55) [reply tsconfig.json:3]: Tests import json files?

Comment (henrikra) [reply tsconfig.json:3]: Alright 👍 In the tests

File: app/routes/postsRouter.ts [hunk 10]
```diff
@@ -1,72 +1,82 @@
...
-
-
-export class PostsRouter {
+//Router to route /posts-route
+class PostsRouter {
     router: Router
+    
 
     constructor() {
         this.router = Router();
```
Comment (henrikra) [inline app/routes/postsRouter.ts]: Is this router still needed? ;)

File: app/routes/postsRouter.ts [hunk 11]
```diff
@@ -1,72 +1,82 @@
 'use strict';
-import {Router, Request, Response, NextFunction, response} from 'express';
-import {Posts} from "../orm/entities/Posts";
+import {Router, Request, Response, NextFunction} from 'express';
+import {Post} from "../orm/entities/Post";
 import {getRepository} from "typeorm";
+import {validate, IsInt, ValidationError} from "class-validator";
```
Comment (henrikra) [inline app/routes/postsRouter.ts]: Hmm double check imports of this file also. Looks like `IsInt` for example is not used

**Review by henrikra — APPROVED** [review]
This is good 👍

**Pull request merged**